packages/vite/src/node/plugins/asset.ts

```diff
@@ -347,6 +347,10 @@ export async function urlToBuiltUrl(
   config: ResolvedConfig,
   pluginContext: PluginContext
 ): Promise<string> {
+  const resolved = await pluginContext.resolve(url, importer, {})
+  if (resolved) {
+    url = path.relative(path.dirname(importer), resolved.id)
+  }
```

Comment:
Let's remove this change so we can move forward with the improved htmlAssetSources. This change may be more involved, better to review it in its own PR.

Reply:
I agree; I had to tweak a _couple_ of things to get the async part to function in some parts of the codebase.